### ISSUE TITLE
add "no logging" option

### ISSUE THREAD

**Issue by CarloLucibello** [PyTorchLightning/pytorch-lightning#645]
I may be wrong, but I see no way to entirely avoid logging during training, which sometimes may be convenient for quick exploratory experiments. 

I suggest to have 
```python
trainer = Trainer(logger=None) 
```
construct a trainer that does no logging at all

**Comment by CarloLucibello** [comment]
wait, actually this avoids logging
```
trainer = Trainer(logger=False, checkpoint_callback=False)
```
If `checkpoint_callback` is not set, a `checkpoints` folder is created in the root directory. 

Two suggestions:
- both arguments should accept the `None` value in addition (or as an alternative) to `False`. It seems more natural to me and the first thing I attempted. 
- Setting `logger=False` should also trigger `checkpoint_callback=False`.

If there is no support for these suggestions the issue can ble closed.

**Comment by awaelchli** [comment]
> Setting logger=False should also trigger checkpoint_callback=False.
Maybe not. The user may want to save checkpoints but no logging turned on. The logger is for experiment tracking (loss curves and other visualizations) and independent of checkpointing.
